slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -319,28 +329,47 @@ bool SLAMServiceImpl::ExtractPointCloudToBuffer(std::stringbuf &buffer) {
 void SLAMServiceImpl::ProcessData() {
     // Set up and build the MapBuilder
     SetUpMapBuilder();
+    DetermineActionMode();
+
+    if (action_mode == SLAMServiceActionMode::UPDATING ||
+        action_mode == SLAMServiceActionMode::LOCALIZING) {
+        // Check if there is an apriori map in the path_to_map directory
+        std::vector<std::string> map_filenames =
+            viam::io::ListSortedFilesInDirectory(path_to_map);
+
+        bool found_map = false;
+        std::string latest_map_filename;
+        for (size_t i = map_filenames.size() - 1; i == 0; i--) {
+            if (map_filenames.at(i).find(".pbstream") != std::string::npos) {
+                latest_map_filename = map_filenames.at(i);
+                found_map = true;
+                break;
+            }
+        }
+        if (!found_map) {
+            throw std::runtime_error(
+                "cannot find maps but they should be present");
+        }
+        bool optimize = true;
```

Comment:
[major] optimize is supposed to be passed in as a config_param. This takes time and there are instances where someone may have already optimized a map on a previous run and so can skip this step

Reply:
That's right, and as soon as we implement the part where we pass in `optimize` via the `config_param`, we can use this here to check whether or not we're going to optimize it. For now it's just a variable set in the code - happy to set it to false if we prefer that. But this implements the framework for letting the user change this flag.

Reply:
Added a comment above the flag pointing to the ticket where it's going to be implemented at.

Reply:
Perfect :)